packages/cli/src/cli/config/action.bundle.assets.ts

```diff
@@ -0,0 +1,102 @@
+import { CommandLineAction, CommandLineStringParameter } from '@rushstack/ts-command-line';
+import { SourceMemory } from '@chunkd/core';
+import { CotarIndexBinary, CotarIndexBuilder, CotarIndexOptions, TarReader } from '@cotar/core';
+import { LogConfig, LogType } from '@basemaps/shared';
+import { promises as fs } from 'fs';
+import { TarBuilder } from '@cotar/tar';
+import { fsa } from '@chunkd/fs';
+
+const Packing = 25; // Packing factor for the hash map
+const MaxSearch = 50; // Max search factor
+
+export class CommandBundleAssets extends CommandLineAction {
+  path: CommandLineStringParameter;
+  output: CommandLineStringParameter;
+
+  public constructor() {
+    super({
+      actionName: 'bundle-assets',
+      summary: 'Cli tool to create cotar file for the config assets',
+      documentation: 'Create a cotar file from a directory',
+    });
+  }
+
+  protected onDefineParameters(): void {
+    this.path = this.defineStringParameter({
+      argumentName: 'ASSETS',
+      parameterLongName: '--assets',
+      description: 'Paths to the input assets files, must contain assets/sprites/ and assets/fonts.',
+    });
+
+    this.output = this.defineStringParameter({
+      argumentName: 'OUTPUT',
+      parameterLongName: '--output',
+      description: 'Paths of the output co tar file.',
+    });
+  }
+
+  protected async onExecute(): Promise<void> {
+    const logger = LogConfig.get();
+    const path = this.path.value;
+    const output = this.output.value;
+    if (path == null || output == null) throw new Error('Please provide a input or ouput path');
+    if (output.endsWith('.tar')) throw new Error(`Invalid output, needs to be .tar.co :"${output}"`);
+
+    logger.info({ indput: path }, 'BundleAssets:Start');
+    const tarFile = await this.buildTar(path, output, logger);
+    const cotarFile = await this.buildTarCo(tarFile, output, logger);
+    logger.info({ output: cotarFile }, 'BundleAssets:Finish');
+  }
+
+  async buildTar(input: string, output: string, logger: LogType): Promise<string> {
+    const startTime = Date.now();
+    const outputTar = output.split('.co')[0];
+
+    const tarBuilder = new TarBuilder(outputTar);
+
+    const files = await fsa.toArray(fsa.list(input));
+    // Ensure files are put into the same order
+    files.sort((a, b) => a.localeCompare(b));
+    logger.info({ output: outputTar, files: files.length }, 'Tar:Create');
+
+    for (const file of files) await tarBuilder.write(file, await fsa.read(file));
```

Comment:
this file name might need to be shortened, what does the tar archive look like?

Reply:
Yes, do we need to be short as `assets/fonts/...` or just `fonts/...`?

Reply:
Updated. By doing this. We also can move the fonts.json to root directory in cotar file.

Reply:
I think just leave the json file as is, and we can just update the server to point at the right spot?

Reply:
It was in `fonts/fonts.json` after build. Then we move it to `fonts.json` in the cotar. Shall we still keep it in the `fonts/font.json`?

Reply:
yeah this bundling logic should replicate the excat file system we should change it in the filesystem if we want it changed!